backend/database-sqlite.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+import { mkdir } from 'node:fs/promises'
+import { join, resolve } from 'node:path'
+import sqlite3 from 'sqlite3'
+
+import { checkKey } from './database.js'
+
+let db: any = null
+let readStatement: any = null
+let writeStatement: any = null
+
+export async function initStorage (options: Object = {}): Promise<void> {
+  const { dirname, filename } = options
+  const dataFolder = resolve(dirname)
+
+  await mkdir(dataFolder, { mode: 0o750, recursive: true })
+
+  await new Promise((resolve, reject) => {
+    if (db) {
+      reject(new Error(`The ${filename} SQLite database is already open.`))
+    }
+    db = new sqlite3.Database(join(dataFolder, filename), (err) => {
+      if (err) {
+        reject(err)
+      }
+    })
+    db.run('CREATE TABLE IF NOT EXISTS Data(key TEXT UNIQUE NOT NULL, value TEXT NOT NULL)', [], (err) => {
```

Comment:
What was the verdict on adding a third column for metadata for dealing with `lastModified` header for fs?

Reply:
That turned out to be a bit complex because `lastModified` has to be stored somewhere and kept up-to-date for all three storage backends, and also because we want a consistent API regardless the data type but only want that field for files.